### PR TITLE
feat: truncate table execute

### PR DESCRIPTION
**Pull request by DevilExileSu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

+ Implement truncate table execute in standalone mode.
+ Implement truncate table execute in distribute mode.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -74,7 +74,7 @@ datafusion-substrait = { git = "https://github.com/waynexia/arrow-datafusion.git
 etcd-client = "0.11"
 futures = "0.3"
 futures-util = "0.3"
-greptime-proto = { git = "https://github.com/MichaelScofield/greptime-proto.git", rev = "71c0002045f6cc8d6e42609bcc5ca6cea963baad" }
+greptime-proto = { git = "https://github.com/DevilExileSu/greptime-proto.git", rev = "c7fea8059be6c5e0b1745d12b0c8304dfde5118d" }
```
Comment (evenyag) [inline Cargo.toml:77]: Could you please also port this to [GreptimeTeam/greptime-proto](https://github.com/GreptimeTeam/greptime-proto)?

File: src/client/src/database.rs [hunk 2]
```diff
@@ -234,6 +234,14 @@ impl Database {
         .await
     }
 
+    pub async fn truncate_table(&self, expr: TruncateTableExpr) -> Result<Output> {
+        let _timer = timer!(metrics::METRIC_GRPC_FLUSH_TABLE);
```
Comment (evenyag) [inline src/client/src/database.rs:238]: Looks like the metric name is unrelated.

File: src/datanode/src/instance/grpc.rs [hunk 3]
```diff
@@ -739,6 +740,35 @@ mod test {
 |   | s |   | 2022-12-30T07:09:00 | 1 |
 +---+---+---+---------------------+---+";
         assert_eq!(recordbatches.pretty_print().unwrap(), expected);
+
+        let query = Request::Ddl(DdlRequest {
```
Comment (evenyag) [inline src/datanode/src/instance/grpc.rs:744]: suggestion: Add a new test for truncate instead of reusing this test.

**Comment by evenyag** [comment]
@WenyXu Please check the changes related to distributed DDL.

File: Cargo.toml (hunk 1 above)
Comment (WenyXu) [reply Cargo.toml:77]: You can create a PR in [GreptimeTeam/greptime-proto](https://github.com/GreptimeTeam/greptime-proto), and just change the `rev` to your PR's head commit without changing the `git`.

**Review by WenyXu — APPROVED** [review]
It doesn't have too many changes related to DDL. Looks good to me.

**Comment by WenyXu** [comment]
BTW, If you're trying to implement a DDL procedure for truncate, consider the way of implementing the idempotent of the truncate.

**Comment by DevilExileSu** [comment]
> BTW, If you're trying to implement a DDL procedure for truncate, consider the way of implementing the idempotent of the truncate.

Got it, Thx!

**Review by evenyag — APPROVED** [review]
LGTM

**Pull request merged**